src/compiler/mcgen.jl

```diff
@@ -7,7 +7,9 @@ function machine(agent::HSAAgent, triple::String)
 
     InitializeAMDGPUTargetMC()
     cpu = get_first_isa(agent) # TODO: Make this configurable
-    feat = "+xnack" # TODO: Make this configurable
+    # TODO: HSA doesn't seem to like +xnack anymore:
+    #feat = "+xnack" # TODO: Make this configurable
```

Comment:
What is xnack?

Reply:
I'm not really sure; it seemed useful at some earlier point but I don't think we've ever made use of it. The most I could find on it (from the LLVM AMDGPU manual) seem to indicate it's some sort of replay feature:

"This is used for demand paging and page migration. If XNACK replay is enabled in the device, then if a page fault occurs the code may execute incorrectly if the xnack feature is not enabled. Executing code that has the feature enabled on a device that does not have XNACK replay enabled will execute correctly, but may be less performant than code with the feature disabled."

https://www.llvm.org/docs/AMDGPUUsage.html#id50

Reply:
Okay let's open an issue about it and delete this code here.

Reply:
Some additional information might be gleaned from this LLVM patch to disable it by default: https://reviews.llvm.org/D40047